fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/EqualTo.java

```diff
@@ -33,6 +33,10 @@ public EqualTo(Expression left, Expression right) {
         super(left, right, "=");
     }
 
+    public static EqualTo of(Expression left, Expression right) {
```

Comment:
What's the benefit of introducing the static method `EqualTo::of`?

Reply:
Like ImmutableList.of()
we don,t need new

Reply:
I think it is not evil to use `new` to instantiate an object. JVM may not always inline some methods. An extra wrapper method which just passes argument to `new` only may introduce some costs.